Content/NPCs/TownModifications/TruffleShopAdditions.cs

```diff
@@ -0,0 +1,25 @@
+﻿using StarlightRiver.Content.Items.Food;
+using StarlightRiver.Content.Items.Utility;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Terraria;
+using Terraria.ID;
+using Terraria.ModLoader;
+using StarlightRiver.Tiles.Mushroom;
+
+namespace StarlightRiver.Content.NPCs.TownModifications
+{
+	class TruffleShopAdditions : GlobalNPC
+	{
+        public override void SetupShop(int type, Chest shop, ref int nextSlot)
+        {
+            if (type == NPCID.Truffle)
```

Comment:
I'm going to let this slide since its likely set up to allow other entries to be added easily